service/controller/v13/encrypter/kms/kms.go

```diff
@@ -0,0 +1,169 @@
+package kms
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/aws-operator/service/controller/v13/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/v13/encrypter"
+	"github.com/giantswarm/aws-operator/service/controller/v13/key"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+type Encrypter struct {
+	logger micrologger.Logger
+
+	installationName string
+}
+
+type EncrypterConfig struct {
+	Logger micrologger.Logger
+
+	InstallationName string
+}
+
+func NewEncrypter(c *EncrypterConfig) (*Encrypter, error) {
+	if c.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", c)
+	}
+
+	if c.InstallationName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.InstallationName must not be empty", c)
+	}
+
+	kms := &Encrypter{
+		logger: c.Logger,
+
+		installationName: c.InstallationName,
+	}
+
+	return kms, nil
+}
+
+func (k *Encrypter) CreateKey(ctx context.Context, customObject v1alpha1.AWSConfig, keyAlias string) error {
+	sc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	key, err := sc.AWSClient.KMS.CreateKey(&kms.CreateKeyInput{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if _, err := sc.AWSClient.KMS.CreateAlias(&kms.CreateAliasInput{
```

Comment:
We only need to use the err value but could we use `if err != nil` here rather than checking inline?

I prefer it style-wise and for me it makes the code a bit easier to read.

Reply:
Good point, done.